services/web/server/src/simcore_service_webserver/login/_2fa.py

```diff
@@ -51,7 +51,7 @@ async def _do_create_2fa_code(
 ) -> str:
     hash_key, code = user_email, generate_passcode()
     await redis_client.set(hash_key, value=code, ex=expiration_seconds)
-    return code
+    return cast(str, code)
```

Comment:
same here.. why can't mypy find this correctly?

Reply:
Ye the same problem here, even if you use str() function in the return value of generate_passcode(). I will continue with other plugins and if I will find out why it is happening will remove the cast.